src/Kentico.Xperience.AzureSearch/Admin/UIPages/BaseIndexEditPage.cs

```diff
@@ -45,47 +68,146 @@ protected async Task<ModificationResponse> ValidateAndProcess(AzureSearchConfigu
 
         if (StorageService.GetIndexIds().Exists(x => x == configuration.Id))
         {
-            var oldIndexConfiguration = StorageService.GetIndexDataOrNull(configuration.Id);
-            var oldIndex = AzureSearchIndexStore.Instance.GetRequiredIndex(oldIndexConfiguration!.IndexName);
+            return await EditIndex(configuration);
+        }
 
-            bool edited = StorageService.TryEditIndex(configuration);
+        return await CreateIndex(configuration);
+    }
 
-            if (edited)
-            {
-                AzureSearchIndexStore.SetIndicies(StorageService);
-                await indexClientService.EditIndex(oldIndex!, configuration, default);
 
-                return new ModificationResponse(ModificationResult.Success);
+    protected static string RemoveWhitespacesUsingStringBuilder(string source)
+    {
+        var builder = new StringBuilder(source.Length);
+
+        for (int i = 0; i < source.Length; i++)
+        {
+            char c = source[i];
+            if (!char.IsWhiteSpace(c))
+            {
+                builder.Append(c);
             }
+        }
+
+        return source.Length == builder.Length ? source : builder.ToString();
+    }
+
+
+    private async Task<ModificationResponse> EditIndex(AzureSearchConfigurationModel configuration)
+    {
+        var oldIndexConfiguration = StorageService.GetIndexDataOrNull(configuration.Id);
+        if (oldIndexConfiguration == null || oldIndexConfiguration.IndexName == null)
+        {
+            eventLogService.LogError(nameof(BaseIndexEditPage), nameof(EditIndex), "The index to edit could not be found.");
+            return new ModificationResponse(ModificationResult.Failure, ["The index to edit could not be found."]);
+        }
 
+        var oldIndex = AzureSearchIndexStore.Instance.GetRequiredIndex(oldIndexConfiguration!.IndexName);
+
+        SearchIndex editedSearchIndex;
+        try
+        {
+            editedSearchIndex = await indexClientService.EditIndex(oldIndex!, configuration, default);
+        }
+        catch (OperationCanceledException ex)
+        {
+            eventLogService.LogError(nameof(BaseIndexEditPage), nameof(EditIndex), $"Index edit operation was cancelled: {ex.Message}");
+            return new ModificationResponse(ModificationResult.Failure);
+        }
+        catch (Exception ex) when (ex is InvalidOperationException or ArgumentNullException or Azure.RequestFailedException)
+        {
+            eventLogService.LogError(nameof(BaseIndexEditPage), nameof(EditIndex), $"Failed to edit Azure Search index: {ex.Message}");
             return new ModificationResponse(ModificationResult.Failure);
         }
 
-        bool created = !string.IsNullOrWhiteSpace(configuration.IndexName) && StorageService.TryCreateIndex(configuration);
+        bool edited = StorageService.TryEditIndex(configuration);
```

Comment:
I believe the proper solution is to open the connection scope by `new CMSConnectionScope()` perform any action inside the using block, commit the transaction if everything goes well, or rollback the transaction in case of any failure/exception.

The same applies also for CreateIndex method.